src/pages/Vault/RequestIssueModal/index.tsx

```diff
@@ -0,0 +1,459 @@
+
+import * as React from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import clsx from 'clsx';
+import {
+  Bitcoin,
+  BitcoinAmount,
+  BitcoinUnit,
+  ExchangeRate,
+  Currency
+} from '@interlay/monetary-js';
+
+import {
+  GovernanceTokenLogoIcon,
+  GOVERNANCE_TOKEN,
+  GOVERNANCE_TOKEN_SYMBOL,
+  WrappedTokenLogoIcon,
+  WRAPPED_TOKEN_SYMBOL
+} from 'config/relay-chains';
+import CloseIconButton from 'components/buttons/CloseIconButton';
+import InterlayModal, {
+  InterlayModalInnerWrapper,
+  InterlayModalTitle
+} from 'components/UI/InterlayModal';
+import { displayMonetaryAmount, getUsdAmount } from 'common/utils/utils';
+import { ParachainStatus, StoreType } from 'common/types/util.types';
+import {
+  newMonetaryAmount,
+  GovernanceUnit,
+  newAccountId,
+  Issue
+} from '@interlay/interbtc-api';
+import TokenField from 'components/TokenField';
+import Big from 'big.js';
+import { BLOCKS_BEHIND_LIMIT } from 'config/parachain';
+import { POLKADOT, KUSAMA } from 'utils/constants/relay-chain-names';
+import STATUSES from 'utils/constants/statuses';
+import { ReactComponent as BitcoinLogoIcon } from 'assets/img/bitcoin-logo.svg';
+import { useErrorHandler } from 'react-error-boundary';
+import { COLLATERAL_TOKEN_ID_LITERAL } from 'utils/constants/currency';
+import PrimaryColorEllipsisLoader from 'components/PrimaryColorEllipsisLoader';
+import SubmittedIssueRequestModal from 'pages/Bridge/IssueForm/SubmittedIssueRequestModal';
+import ErrorModal from 'components/ErrorModal';
+import Hr2 from 'components/hrs/Hr2';
+import SubmitButton from 'components/SubmitButton';
+import InformationTooltip from 'components/tooltips/InformationTooltip';
+import PriceInfo from 'pages/Bridge/PriceInfo';
+
+const WRAPPED_TOKEN_AMOUNT = 'amount';
+const BTC_ADDRESS = 'btc-address';
+
+type RequestIssueFormData = {
+  [WRAPPED_TOKEN_AMOUNT]: string;
+  [BTC_ADDRESS]: string;
+}
+
+interface Props {
+  onClose: () => void;
+  open: boolean;
+  vaultAddress: string;
+}
+
+let EXTRA_REQUIRED_COLLATERAL_TOKEN_AMOUNT: number;
+if (process.env.REACT_APP_RELAY_CHAIN_NAME === POLKADOT) {
+  EXTRA_REQUIRED_COLLATERAL_TOKEN_AMOUNT = 0.2;
+} else if (process.env.REACT_APP_RELAY_CHAIN_NAME === KUSAMA) {
+  EXTRA_REQUIRED_COLLATERAL_TOKEN_AMOUNT = 0.01;
+} else {
+  throw new Error('Something went wrong!');
+}
+const extraRequiredCollateralTokenAmount =
+  newMonetaryAmount(EXTRA_REQUIRED_COLLATERAL_TOKEN_AMOUNT, GOVERNANCE_TOKEN, true);
+
+// TODO: share form with bridge page
+const RequestIssueModal = ({
```

Comment:
This component is based on `pages/Bridge/IssueForm` component. It will be possible to significantly simplify it once the IssueForm is refactored in a reusable way.